content/ros/ROS_Nodes_Topics_Messages/presentation/ros_nodes_topics_messgaes.tex

```diff
@@ -0,0 +1,1560 @@
+\documentclass{beamer}
+
+
+\usepackage[utf8]{inputenc}
+\usepackage{amsmath}
+\usepackage{amsfonts}
+\usepackage{amssymb}
+\usepackage{graphicx}
+\usepackage{ragged2e}  % `\justifying` text
+\usepackage{booktabs}  % Tables
+\usepackage{tabularx}
+\usepackage{tikz}      % Diagrams
+\usetikzlibrary{calc, shapes, backgrounds}
+\usepackage{amsmath}
+\usepackage{amssymb}
+\usepackage{dsfont}
+\usepackage{url}       % `\url
+\usepackage{listings}  % Code listings
+\usepackage[T1]{fontenc}
+\usepackage[percent]{overpic}
+\usetikzlibrary{trees}
+\usepackage[absolute,overlay]{textpos}
+\usepackage{tcolorbox}
+
+
+
+\newtcolorbox{terminal}{colback=black!70!white,colframe=black!70!white}
+\newtcolorbox{focus}{colback=black!10!white,colframe=black!10!white}
+\newtcolorbox{terminal2}[1]{colback=white,colframe=black!70!white,fonttitle=\bfseries,title=#1}
+
+\usepackage{theme/beamerthemehbrs}
+
+\author[MAS]{Hassan Umari}
+\title{ROS Nodes, Topics, and Messages}
+\subtitle{Foundation Course}
+\institute[HBRS]{Hochschule Bonn-Rhein-Sieg}
+\date{\today}
+\subject{ROS workshop}
+
+% \thirdpartylogo{path/to/your/image}
+
+
+\begin{document}
+{
+\begin{frame}
+\titlepage
+\end{frame}
+}
+
+
+\section{Recap}
+\begin{frame}{Recap}
+    \framesubtitle{Summary of yesterday's session}
+    \begin{itemize}
+        \item ROS is a collection of libraries and tools that helps you when you develop software for robots.
+              
+        \item ROS provides several ways to transfer data between nodes:
+        
+        \begin{enumerate}
+            \item ROS topics and messages (\textbf{publish/subscribe}).
+            \item ROS services (\textbf{request/reply}).
+            \item ROS actions (\textbf{request/reply}).
+            \item Parameter server.
+        \end{enumerate}   
+    \end{itemize}
+\end{frame}
+
+
+\begin{frame}{Recap}
+    \framesubtitle{Summary of yesterday's session}
+    \begin{itemize}
+        \item We will focus today on ROS topics and messages..
```

Comment:
Today, we will focus on ROS topics and messages.

Reply:
done